common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -81,8 +82,14 @@
     private static final String TAG = CommandDispatcher.class.getSimpleName();
 
     private static final int SILENT_REQUEST_THREAD_POOL_SIZE = 5;
-    private static final ExecutorService sInteractiveExecutor = Executors.newSingleThreadExecutor();
-    private static final ExecutorService sSilentExecutor = Executors.newFixedThreadPool(SILENT_REQUEST_THREAD_POOL_SIZE);
+    private static final int INTERACTIVE_REQUEST_THREAD_POOL_SIZE = 1;
+    //TODO:1315931 - Refactor the threadpools to not be unbounded for both silent and interactive requests.
+    private static final ExecutorService sInteractiveExecutor = ThreadUtils.getNamedThreadPoolExecutor(
+            1, INTERACTIVE_REQUEST_THREAD_POOL_SIZE, -1, 0, TimeUnit.MINUTES, "interactive"
+    );
+    private static final ExecutorService sSilentExecutor = ThreadUtils.getNamedThreadPoolExecutor(
```

Comment:
@iambmelt tagging you as a FYI, went over this with @AdamBJohnsonx

Reply:
thx @tanmaymanolkar1 lgtm